### PR TITLE
Implement translations from E2DR to DR2E

### PR DESCRIPTION
**Pull request by gasperzgonec**
- **Added isEventType function from devrev-adaas-typescript.**
- **Added a function for event type translation and added mapping from old names to the new ones.**

## Summary

Migrated event type names from old E2DR to DR2E naming convention.
Added a simple function for easier translation and migration.

## Connected Issues

- [#ISS-131089](https://app.devrev.ai/devrev/works/ISS-131089)

## Craftsmanship, Integrity, and Devil’s Advocacy

- [ ] Testing: Negative test cases: null or default values, crash and fault injection tests
- [ ] Testing: Boundary conditions: rolling upgrades, denial-of-service, etc.
- [ ] Testing: Fixing a few existing — flaky or permanently-broken — test cases
- [ ] Observing: Detailed error codes so machines can understand
- [ ] Observing: Adding superior metrics for future debugging
- [ ] Observing: Tracing the hairiest pathways for field serviceability
- [ ] Training: KnowledgeOps update? So AI always works.

### CODE REVIEW

File: src/common/helpers.ts [hunk 1]
```diff
@@ -14,6 +14,63 @@ import { readFileSync } from 'fs';
...
+
+export function isEventType({
+  event,
+  eventType,
+}: {
+  event: AirdropEvent;
+  eventType: EventType;
+}): boolean {
+  return translateEventType(event.payload.event_type) === translateEventType(eventType);
+}
```
Comment (patricijabrecko) [inline src/common/helpers.ts:72]: Thinking from the perspective that most ADaaS developers started from our template and we want them to need to do as little changes as possible in their code, I suggest instead of creating `isEventType` function to create `getEventType`, where they would put in event or `event.payload.event_type` and get the right translation.

I thought about it looking at this code section that every snap-in uses:

```ts
switch (event.payload.event_type) { // Developer would need to change this to getEventType(event.payload.event_type)
    case EventType.ExtractionExternalSyncUnitsStart:
      path = __dirname + '/workers/external-sync-units-extraction';
      break;
    case EventType.ExtractionMetadataStart:
      path = __dirname + '/workers/metadata-extraction';
      break;
    case EventType.ExtractionDataStart:
    case EventType.ExtractionDataContinue:
      path = __dirname + '/workers/data-extraction';
      break;
    case EventType.ExtractionAttachmentsStart:
    case EventType.ExtractionAttachmentsContinue:
      path = __dirname + '/workers/attachments-extraction';
      break;
  }
```

What do you think?

File: src/common/helpers.ts [hunk 2]
```diff
@@ -14,6 +14,63 @@ import { readFileSync } from 'fs';
...
+  "EXTRACTION_DATA_DELETE_DONE": ExtractorEventType.ExtractionDataDeleteDone,
+  "EXTRACTION_DATA_DELETE_ERROR": ExtractorEventType.ExtractionDataDeleteError,
+  "EXTRACTION_ATTACHMENTS_DELETE_DONE": ExtractorEventType.ExtractionAttachmentsDeleteDone,
+  "EXTRACTION_ATTACHMENTS_DELETE_ERROR": ExtractorEventType.ExtractionAttachmentsDeleteError
+};
+
+/**
+ * Translates Event type from the old naming scheme to the new one
+ */
+export function translateEventType(event_type: string): EventType | ExtractorEventType {
```
Comment (patricijabrecko) [inline src/common/helpers.ts:48]: Let's use camelCase for variables (event_type) to be more consistent throughout the SDK

**Review by patricijabrecko — COMMENTED** [review]
@GasperSenk we need to wait for changes in the snap-in manager before we can promote this right?

File: src/common/helpers.ts (hunk 1 above)
Comment (gasperzgonec) [reply src/common/helpers.ts:72]: That sound like a nice solution. I wasn't entirely sure where this code would be used, but this gives me a nice starting point :)
Will fix.

File: src/types/extraction.ts [hunk 3]
```diff
@@ -39,22 +39,22 @@ export enum EventType {
...
+  ExtractionExternalSyncUnitsError = 'EXTERNAL_SYNC_UNIT_EXTRACTION_ERROR',
+  ExtractionMetadataDone = 'METADATA_EXTRACTION_DONE',
+  ExtractionMetadataError = 'METADATA_EXTRACTION_ERROR',
+  ExtractionDataProgress = 'DATA_EXTRACTION_PROGRESS',
+  ExtractionDataDelay = 'DATA_EXTRACTION_DELAYED',
+  ExtractionDataDone = 'DATA_EXTRACTION_DONE',
+  ExtractionDataError = 'DATA_EXTRACTION_ERROR',
+  ExtractionAttachmentsProgress = 'ATTACHMENT_EXTRACTION_PROGRESS',
+  ExtractionAttachmentsDelay = 'ATTACHMENT_EXTRACTION_DELAYED',
+  ExtractionAttachmentsDone = 'ATTACHMENT_EXTRACTION_DONE',
```
[the remark this answers is not in the document]
Comment (gasperzgonec) [reply src/types/extraction.ts:52]: Nice catch!
Fixed.

File: src/common/helpers.ts [hunk 4]
```diff
@@ -14,6 +14,53 @@ import { readFileSync } from 'fs';
...
+  "EXTRACTION_DATA_DELETE_DONE": ExtractorEventType.ExtractionDataDeleteDone,
+  "EXTRACTION_DATA_DELETE_ERROR": ExtractorEventType.ExtractionDataDeleteError,
+  "EXTRACTION_ATTACHMENTS_DELETE_DONE": ExtractorEventType.ExtractionAttachmentsDeleteDone,
+  "EXTRACTION_ATTACHMENTS_DELETE_ERROR": ExtractorEventType.ExtractionAttachmentsDeleteError
+};
+
+/**
+ * Translates Event type from the old naming scheme to the new one
+ */
+export function getEventType(eventType: string): EventType | ExtractorEventType {
```
Comment (radovanjorgic) [inline src/common/helpers.ts:48]: How this will be used? Do we need to use this function wherever we are checking event type? If yes, can we avoid that somehow?

**Review by radovanjorgic — CHANGES_REQUESTED** [review]

File: src/common/helpers.ts (hunk 4 above)
Comment (gasperzgonec) [reply src/common/helpers.ts]: This was a massive oversight on my side.

I've now added it to the top of the `spawn` function, so that every event that comes into the Lambda is translated and reused as the translated one everywhere.

File: src/common/helpers.ts (hunk 2 above)
Comment (gasperzgonec) [reply src/common/helpers.ts]: Fixed :)

File: src/common/helpers.ts [hunk 5]
```diff
@@ -14,6 +14,61 @@ import { readFileSync } from 'fs';
...
+  if(event_type in EVENT_TYPE_TRANSLATION_TABLE){
+    return EVENT_TYPE_TRANSLATION_TABLE[event_type as keyof typeof EVENT_TYPE_TRANSLATION_TABLE];
+  }
+
+  // Return the correct event type
+  if (event_type in ExtractorEventType) {
+    return event_type as ExtractorEventType;
+  }
+
+  return event_type as EventType;
```
[the remark this answers is not in the document]
Comment (gasperzgonec) [reply src/common/helpers.ts]: I have added the `UNKNOWN_EVENT_TYPE` to the `EventType` so that it can now be returned.

**Review by GasperSenk — APPROVED** [review]

**Comment by GasperSenk** [comment]
Lets write down from which version of the SDK, the SDK will accept new event types.

It can start returning new event types now as the gateway is already translating them.

File: src/workers/spawn.ts [hunk 6]
```diff
@@ -93,6 +97,7 @@ export async function spawn<ConnectorState>({
   initialDomainMapping,
   options,
 }: SpawnFactoryInterface<ConnectorState>): Promise<void> {
+  event.payload.event_type = getEventType(event.payload.event_type);
```
Comment (radovanjorgic) [inline src/workers/spawn.ts]: Did you check if this is the only place where we need to do this translation?
Also, do you think it is worth creating some `transformEvent` function which transforms the whole event in new event with some fields edited or structure changed a bit (in case we need it in future)?

File: src/common/helpers.ts [hunk 7]
```diff
@@ -19,6 +19,43 @@ import {
...
+/**
+ * Translates Event type from the old naming scheme to the new one
+ * 
+ * @param eventType - The event type string to translate
+ * @returns EventType - The translated event type with the following behavior:
+ *   1) Old E2DR names are translated to new DR2E format
+ *   2) Valid DR2E names are returned as-is
+ *   3) Unknown values return `UnknownEventType`
+ */
+export function getEventType(eventType: string): EventType {
```
Comment (radovanjorgic) [inline src/common/helpers.ts]: I would rename this to sth like transform/translateEventType, since we already have event type but need to translate it to the new one.
